torch/_dynamo/variables/functions.py

```diff
@@ -9,37 +9,58 @@
 from .. import variables
 from ..bytecode_transformation import create_instruction
 from ..exc import unimplemented
-from ..source import AttrSource, ConstantSource, GetItemSource
-from ..utils import make_cell
+from ..source import AttrSource, ConstantSource, DefaultsSource, GetItemSource
+from ..utils import istensor, make_cell
 from .base import typestr, VariableTracker
 
 
-def wrap_bound_arg(val, options):
+def wrap_bound_arg(tx, val, options, source=None):
+    # Source propagation is best effort since not every object we encounter has a source to begin with.
+    assert (
+        "source" not in options
+    ), "Source needs to be separate from options due to recursive calls for lists/dicts"
+
     if isinstance(val, dict):
         return variables.ConstDictVariable(
-            {k: wrap_bound_arg(v, options) for k, v in val.items()}, dict, **options
+            {
+                k: wrap_bound_arg(tx, v, options, source=getattr(v, "source", None))
+                for k, v in val.items()
+            },
+            dict,
+            **options,
         )
     elif isinstance(val, (tuple, list)):
         cls = variables.BaseListVariable.cls_for(type(val))
-        return cls([wrap_bound_arg(x, options) for x in val], **options)
-    elif variables.ConstantVariable.is_literal(val):
+        return cls(
+            [
+                wrap_bound_arg(tx, x, options, source=getattr(x, "source", None))
+                for x in val
+            ],
+            **options,
+        )
+
+    if variables.ConstantVariable.is_literal(val):
         return variables.ConstantVariable(val, **options)
     elif isinstance(val, types.FunctionType):
-        return variables.UserFunctionVariable(val, **options)
+        return variables.UserFunctionVariable(val, source=source, **options)
     elif isinstance(val, enum.Enum):
-        return variables.EnumVariable(val, **options)
+        return variables.EnumVariable(val, source=source, **options)
     elif isinstance(val, (type, abc.ABCMeta)):
-        return variables.UserDefinedClassVariable(val, **options)
+        return variables.UserDefinedClassVariable(val, source=source, **options)
+    elif istensor(val):
+        from torch._dynamo.variables.builder import VariableBuilder
+
+        return VariableBuilder(tx, source=source, **options)(val)
```

Comment:
This doesn't make much sense to me, VariableBuilder doesn't actually take any arguments besides tx and source...